### PR TITLE
Add a guide on prebuilds

### LINKED ISSUE #261: Improve documentation of dev container image prebuild and update templates
Many developers want to understand how to enable the reuse of an existing container image each time a session is started, as opposed to rebuilding the container from scratch so that:

- Create a codified build event for deterministic environment definition
- Speed environment creation
- In the Codespaces Prebuilds case, work around the limitation that Codespaces do not follow a fork by

We documentation [prebuilding dev container images](https://containers.dev/implementors/reference/#prebuilding), but none of our templates illustrate good practices for this, and the documentation is a but buried. 

I propose that we create a [guide](https://containers.dev/guides) that describes how to decide if you need to prebuild, walk through the mechanics on GitHub with Actions and GHCR, and provides a few good examples.
It would also help if we update the templates where a prebuilt image would help to illustrate the pattern.

### PR DESCRIPTION
**Pull request by bamurtaugh**
Fixes https://github.com/devcontainers/spec/issues/261. 

@craiglpeters I updated the info on the Kubernetes example a bit based on our chat last week - please feel free to propose other changes, or add in other tips/feedback from your experience working on the prebuild!

I also updated post.html to hopefully allow for 2 author posts correctly - cc @joshspicer in case you have any feedback on a different way to achieve this.

cc @olguzzar thanks again for the feedback - I'd love an additional review as I've incorporated your feedback and made a few more updates! I also created https://aka.ms/dev-container-prebuild that'll point to this guide for us to use in any docs or advocacy work.

### CODE REVIEW

**Review by samruddhikhandale — DISMISSED** [review]
This is awesome and pretty helpful!
Thank you, Brigit and Craig 👏 ✨

**Comment by joshspicer** [comment]
I opened this PR up in a codespace (with `bundle exec jeykll serve`) and i'm not seeing the authors anymore 🤔
image

**Comment by joshspicer** [comment]
Discussed my comment on authors with Brigit offline :)

**Review by joshspicer — DISMISSED** [review]

File: _posts/2023-08-22-prebuild.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+* Any time a change is made to the dev container, the repo currently uses the dev container [GitHub Action](https://github.com/craiglpeters/kubernetes-devcontainer/actions/workflows/devcontainer-build-and-push.yml) to build the image and push it to GHCR
+     * You can check out its latest prebuilt image in the [`Packages` tab](https://github.com/users/craiglpeters/packages/container/package/kubernetes-devcontainer) of its GitHub Repo. In this tab, you can see its GHCR URL is `ghcr.io/craiglpeters/kubernetes-devcontainer:latest`
+* The main Kubernetes repo and any fork of it can now define a `.devcontainer` folder and [reference this prebuilt image](https://github.com/craiglpeters/kubernetes-devcontainer/blob/master/.devcontainer/devcontainer.json#L7) through: `"image": "ghcr.io/craiglpeters/kubernetes-devcontainer:latest"`
+
+**Dev container spec images**
+* This repo prebuilds a variety of dev containers, each of which is defined in their individual folks in the [src folder](https://github.com/devcontainers/images/tree/main/src)
```
Comment (olguzzar) [inline _posts/2023-08-22-prebuild.md]: should this say "in their individual **folders**"?

**Comment by olguzzar** [comment]
This looks so great Brigit!!

File: _posts/2023-08-22-prebuild.md (hunk 1 above)
Comment (bamurtaugh) [reply _posts/2023-08-22-prebuild.md]: It definitely should 😄 updating now

Comment (bamurtaugh) [reply _posts/2023-08-22-prebuild.md]: Pushed a fix - thanks!

**Comment by bamurtaugh** [comment]
Incorporated feedback from yesterday's discussion (cross-link to this post from Chuck's guide on images vs dockerfile, added info on multi-stage builds). Would love another quick set of reviews, then I think we can merge this in and of course keep iterating in the future!

File: _posts/2022-12-16-dockerfile.md [hunk 2]
```diff
@@ -44,7 +46,7 @@ That's it! When you start up your Dev Container, the Dockerfile will be automati
 
 Better yet, you can can use a Dockerfile as a part of authoring an image you can share with others. You can even **add Dev Container settings and metadata right into the image itself**. This avoids having to duplicate config and settings in multiple devcontainer.json files and keeps them in sync with your images! 
 
-See the reference on **[pre-building](/implementors/reference/#prebuilding)** to learn more!
+See the guiide on **[pre-building](/_posts/2023-08-22-prebuild.md)** to learn more!
```
Comment (chrmarti) [inline _posts/2022-12-16-dockerfile.md]: `guide`

**Review by chrmarti — DISMISSED** [review]
Looks great. Maybe a separate sample repo just showing off prebuilds would makes it easier for users to get an overview of what is needed. Not a requirement though.

File: _posts/2022-12-16-dockerfile.md (hunk 2 above)
Comment (bamurtaugh) [reply _posts/2022-12-16-dockerfile.md]: Thanks!

**Comment by bamurtaugh** [comment]
Thanks so much @chrmarti! That's great feedback, and I've opened a tracking issue (in case anyone from the community happens to have one they'd like to contribute too!): https://github.com/devcontainers/devcontainers.github.io/issues/281.

File: _posts/2023-08-22-prebuild.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+     npm install -g @devcontainers/cli
+     ```
+
+* Build your image and push it to a container registry (like the [Azure Container Registry](https://learn.microsoft.com/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli), [GitHub Container Registry](https://docs.github.com/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images), or [Docker Hub](https://docs.docker.com/engine/reference/commandline/push)):
+
+     ```bash
+     devcontainer build --workspace-folder . --push true --image-name <my_image_name>:<optional_image_version>
+     ```
+
+* You can automate pre-building your image by scheduling the build using a DevOps or continuous integration (CI) service like GitHub Actions. We've created a [GitHub Action](https://github.com/marketplace/actions/devcontainers-ci) and [Azure DevOps task](https://marketplace.visualstudio.com/items?itemName=devcontainers.ci) to help with this.
```
Comment (markphip) [inline _posts/2023-08-22-prebuild.md:65]: The link here to the GitHub Action does not work. Maybe it is not published in the Marketplace?

**Review by chrmarti — APPROVED** [review]

**Review by samruddhikhandale — APPROVED** [review]

File: _posts/2023-08-22-prebuild.md (hunk 3 above)
Comment (bamurtaugh) [reply _posts/2023-08-22-prebuild.md:65]: Thanks for flagging - I'll merge this PR and then open a separate thread for that.

**Pull request merged**